habitat_sim/utils/datasets_download.py

```diff
@@ -0,0 +1,249 @@
+        return
+    root_path = data_sources[uid]["root"]
+    print(f"Cleaning datasource ({uid}) directory: {root_path}")
+    try:
+        shutil.rmtree(root_path)
+    except OSError as e:
+        print(f"Error: {dir_path} : {e.strerror}")
+
+
+def download_and_place(uid, replace=False):
+    r"""Data-source download function. Validates uid, handles existing data version, downloads data, unpacks, writes version, cleans up."""
+    if not data_sources.get(uid):
+        print(f"Data download failed, no datasource named {uid}")
+        return
+
+    root_path = data_sources[uid]["root"]
+    version_tag = data_sources[uid]["version"]
+    # check for current version
+    version_filepath = os.path.join(
+        data_path,
+        data_sources[uid]["root"] + "DATA_VERSION_" + version_tag,
+    )
+    if os.path.exists(data_sources[uid]["root"]):
+        resolved_existing = False
+        replace_existing = False
+        for file in os.listdir(root_path):
+            if "DATA_VERSION" in file:
+                existing_version = file[13:]
+                if existing_version == version_tag:
+                    print(
+                        f"Existing data source ({uid}) version ({existing_version}) is current, aborting download. Root directory: {root_path}"
+                    )
+                    return
+                else:
+                    # found root directory with different version
+                    found_version_message = f"({uid}) Found previous data version ({existing_version}). Replace with requested version ({version_tag})?"
+                    replace_existing = (
+                        replace if replace else prompt_yes_no(found_version_message)
+                    )
+                    resolved_existing = True
+        if not resolved_existing:
+            # found root directory with unspecified version
+            found_version_message = f"({uid}) Found unknown data version. Replace with requested version ({version_tag})?"
+            replace_existing = (
+                replace if replace else prompt_yes_no(found_version_message)
+            )
+        if replace_existing:
+            clean_data(uid)
+        else:
+            print("Not replacing data, aborting download.")
+            return
+
+    # download
+    download_pre_args = data_sources[uid].get("download_pre_args", "")
+    download_post_args = data_sources[uid].get("download_post_args", "")
+
+    download_command = (
+        "wget --continue "
+        + download_pre_args
+        + data_sources[uid]["source"]
+        + " -P "
+        + data_path
+        + download_post_args
+    )
+    # print(download_command)
+    os.system(download_command)
+    assert os.path.exists(
+        os.path.join(data_path, data_sources[uid]["package_name"])
+    ), "Download failed, no package found."
+
+    # unpack
+    package_name = data_sources[uid]["package_name"]
+    if package_name.endswith(".zip"):
```

Comment:
Tar has an almost identical API in Python too. Might as well just add it in here too: https://docs.python.org/3/library/tarfile.html#tarfile.open